### PR TITLE
Added a delete button in EditEventModal for admin and senders.

### PR DESCRIPTION
**Pull request by danayim03**
#10: Added a "Delete Event" button within the EditEventModal, allowing admins and senders to delete events. When the button is clicked, a confirmation modal appears displaying the event's title, asking the user to confirm the deletion. Upon confirmation, an alert modal appears to notify the user that the event has been successfully deleted, again showing the event's title for clarity.

*The error case for deleteEventHandler has not been manually tested yet. For now the response always returns ok.

Screenshot 2024-10-21 at 2 31 59 AM
Screenshot 2024-10-21 at 2 32 41 AM
Screenshot 2024-10-21 at 2 32 30 AM

### CODE REVIEW

File: app/redux/modalSlice.tsx
Comment (lad6558) [inline app/redux/modalSlice.tsx:1]: Remove unused code here to avoid confusion for future maintainers

File: app/components/Modal.tsx
Comment (lad6558) [inline app/components/Modal.tsx:1]: Remove comments that won't be useful for other people

File: app/components/EditEventModal.tsx
Comment (lad6558) [inline app/components/EditEventModal.tsx:63]: A little nitpicking: has this case been tested? In what case will this response be not ok? If it has been tested, also include the possible situation in your PR description. Otherwise maybe point out here that for now the response always returns ok

File: app/api/delete-event/route.ts
Comment (lad6558) [inline app/api/delete-event/route.ts]: Maybe instead of comments at the top of the file, these should be separated into doc comments for individual functions

File: app/api/delete-event/route.ts
Comment (lad6558) [inline app/api/delete-event/route.ts]: (Just to correct any misunderstanding you might have), I think `console.error` here will only prints out to the `npm run dev` console (i.e. the developer), because backend code only prints out to the terminal where frontend code prints out to the browser console.

File: app/api/delete-event/route.ts
Comment (lad6558) [inline app/api/delete-event/route.ts]: This is a minor detail here, but while this `throw` statement will indeed stop the execution of the `deleteEvent` function, but it's important to note that it doesn't automatically stop the execution of the entire API route. The error will be propagated to the caller (the DELETE function), but it's not being caught or handled there.

Consider adding a try/catch block in the DELETE function to handle this error gracefully. 

```js
export async function DELETE(request: Request) {
// ... existing code ...
  try {
    await deleteEvent(id);
    return NextResponse.json({ success: true });
    } catch (error) {
    console.error(Error in DELETE route: ${error});
    return NextResponse.json("Internal server error", { status: 500 });
    }
}
```

This way, if an error occurs during the deletion process, we can return an appropriate error response to the client instead of potentially crashing the server or leaving the request hanging. This also makes the `response.ok` block down the line make more sense.

**Review by lad6558 — CHANGES_REQUESTED** [review]
This is pretty great for a first time contributor. The delete button works. I also like the touch to have the event title in the alert. Great PR description too - descriptive title, detailed description, screenshots.  Great job Dana!

I just have some comments in regard to coding style. You might want to document your testing for error-handling behaviors (this is not super important here, because `delete` is not that error prone, but it is important to have a good test coverage over your code). Also remember to remove remove unused empty files such as `Icon` and `app/components/Trashcan.tsx` and unused code that is commented out (those related to the modal) to give future maintainers an easier time reading the code.

File: app/components/EditEventModal.tsx
Comment (danayim03) [reply app/components/EditEventModal.tsx:63]: There would be couple of cases when the response is not ok: network error / server-side issues / incorrect endpoint / unauthorized access, etc (I asked chatgpt about this btw). 

The fact that the delete works shows that there are no server-side issues nor incorrect endpoint (correct me if im wrong). But I have not tested the case when an unauthorized person tries to delete an event. How would I usually test this? Do I need a random mit student's account who is not authorized to delete events?

File: app/api/delete-event/route.ts
Comment (danayim03) [reply app/api/delete-event/route.ts]: or can I just get rid of the comments since the team members won't need them? I had them just for me to have a better understanding of what the functions did when I was writing them.

File: app/api/delete-event/route.ts
Comment (danayim03) [reply app/api/delete-event/route.ts]: oh you're right- thx for pointing it out. I guess since it's obvious that the logs in route.ts are for developers, I will remove these comments instead of changing my comment.

Comment (gabrc52) [reply app/api/delete-event/route.ts]: I think it's fine to leave the logging for developers, because if things do go wrong for some reason, it will be easier to know what the error was from looking at the logs without having to reproduce the issue.

Edit ah nvm yeah you are removing the comment not the logging

Comment (danayim03) [reply app/api/delete-event/route.ts]: yea I will leave the logs but I will delete the comments

File: app/components/EditEventModal.tsx
Comment (gabrc52) [reply app/components/EditEventModal.tsx:63]: You can test the allowDelete function, but @lad6558 it seems that this repository does not have any tests?

I think it is always worth it to manually check that people who shouldn't have access aren't allowed to perform operations. Sometimes even large companies forget

Comment (lad6558) [reply app/components/EditEventModal.tsx:63]: Yeah makes sense - if it is difficult to test, no need to actually test it. I would prefer the fact that this error case has not been tested to be in the comment, but that is a rather minor style preference. 

@gabrc52 Yeah I am talking about manual testing. Thanks for the comment

File: app/redux/modalSlice.tsx
Comment (lad6558) [inline app/redux/modalSlice.tsx]: Is there a event-delete modal?

**Review by lad6558 — APPROVED** [review]
I believe there is some redundant code in modalSlice.tsx. After you delete it, feel free to merge

File: app/redux/modalSlice.tsx
Comment (danayim03) [reply app/redux/modalSlice.tsx]: Oh no. I forgot to delete that - I will delete it. That was written when Andres suggested a delete modal but you didn't like the idea

**Pull request merged**
